examples/mcuboot_ota/Kconfig

```diff
@@ -0,0 +1,63 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+menuconfig EXAMPLES_MCUBOOT_OTA
+	tristate "MCUboot OTA example"
+	default n
+	select BOOT_MCUBOOT
+	---help---
+		Example application demonstrate a software firmware update.
+		It downloads an application firmware image from a given URL.
+
+if EXAMPLES_MCUBOOT_OTA
+
+comment "This example requires network"
+	depends on !NET_TCP
+
+choice EXAMPLES_MCUBOOT_OTA_APP
+	prompt "Select MCUboot application"
+	default EXAMPLES_MCUBOOT_OTA_APP_AGENT
+	depends on NET_TCP
+	---help---
+		Select MCUboot application.
+
+config EXAMPLES_MCUBOOT_OTA_APP_AGENT
+	bool "Agent"
+	---help---
+		Implements the update agent that downloads the application
+		firmware image from a given URL and saves it to the secondary
+		slot as a pending update.
+
+config EXAMPLES_MCUBOOT_OTA_APP_CONFIRM
+	bool "Confirm"
+	---help---
+		Is the application to be download to demonstrate firmware
+		upgrade functionality.
+
+endchoice
```

Comment:
What's the reason for making this examples mutually-exclusive?

Reply:
This example require two steps: one is the download agent, second is the app that will be used to demonstrate that application (agent) was upgraded, which it should be different at some aspect to achieve that. My understanding is that it require two different build.

Reply:
I believe this is an unnecessary restriction.
One may create an application firmware image that includes both examples and create a testing loop performing update-confirm-repeat using the same base image.